File: BOJ_Swift/트리/BOJ_4358_생태학.swift
//
//  main.swift
//  Algorithm
//
//  Created by GYURI PARK on 2023/11/28.
//

import Foundation

var trees = [String: Int]()
var cnt = 0

while let tree = readLine(), !tree.isEmpty {
    cnt += 1
    if let count = trees[tree] {
        trees[tree] = count + 1
    } else {
        trees[tree] = 1
    }
}

var sorts = trees.sorted{$0.0 < $1.0}

for i in sorts {
    let per = (Double(i.value) / Double(cnt) * 100)
    print("\(i.key) \(String(format: "%.4f", per))")
}

File: BOJ_Python/트리/BOJ_4358_생태학.py
import sys

trees = dict()
cnt = 0
while True:
    tree = sys.stdin.readline().rstrip()
    if not tree:
        break
    cnt += 1
    if tree in trees:
        trees[tree] += 1
    else:
        trees[tree] = 1
    
sorts = dict(sorted(trees.items()))
for i in sorts:
    per = (sorts[i] / cnt * 100)
    print("%s %.4f" %(i, per))
